### PR TITLE
Dependancy/Nodejs version updates

### PR DESCRIPTION
**Pull request by cabbiepete**
Updated to work with latest versions of dependancies, random-lib, async etc. and also nodes 4.x and above.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "xkcd-password",
-  "version": "1.2.0",
+  "version": "2.0.0",
```
Comment (fardog) [inline package.json]: please revert this line; i prefer to bump versions with `npm version major` on the CLI for the auto git tagging and etc.

**Review by fardog — CHANGES_REQUESTED** [review]
thanks tons for this PR! i'd been meaning to get around to this!

just one nit to correct please; after that's done I'll merge this and release as v2.0.0.

thank you!

File: package.json (hunk 1 above)
Comment (cabbiepete) [inline package.json]: Ok sure that's exactly what I did but I guess the tag does not come through with the pull request.

**Review by cabbiepete — COMMENTED** [review]
Done

**Review by fardog — APPROVED** [review]

**Comment by cabbiepete** [comment]
Thanks for the quick response.

**Comment by fardog** [comment]
no problem; thank you for the pr! version 2.0.0 has been published.

**Pull request merged**